bee-plugin/src/event.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Types for events.
+
+pub use crate::grpc::{DummyEvent, SillyEvent};
+
+use std::convert::TryFrom;
+
+/// Identifier for each event type/
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+#[repr(u8)]
+pub enum EventId {
+    /// Identifier for `DummyEvent`.
+    Dummy = 0,
+    /// Identifier for `SillyEvent`.
+    Silly = 1,
+}
+
+impl TryFrom<u8> for EventId {
```

Comment:
Depending on how long this enum will get eventually it might make sense to derive `num::FromPrimitive` for this enum.

Reply:
I think it will be pretty long.